src/vos/vos_obj.c

```diff
@@ -1253,6 +1408,19 @@ vos_obj_iter_fini(struct vos_iterator *iter)
 	    (iter->it_type == VOS_ITER_DKEY || !iter->it_from_parent))
 		vos_obj_release(vos_obj_cache_current(), oiter->it_obj);
 
+	switch (iter->it_type) {
+	case VOS_ITER_SINGLE:
+	case VOS_ITER_RECX:
+		if (iter->it_from_parent)
+			break;
```

Comment:
What is the purpose for such check?

Reply:
hmm, not sure what this is for...probably some code that needs to be cleaned up.